### PR TITLE
Support weight only quantization with intel-extension-for-transformers.

### PR DESCRIPTION
**Pull request by PenghuiCheng**
Support weight only quantization with intel-extension-for-transformers.
[Intel® Extension for Transformers](https://github.com/intel/intel-extension-for-transformers) is an innovative toolkit to accelerate Transformer-based models on Intel platforms, in particular effective on 4th Intel Xeon Scalable processor [Sapphire Rapids](https://www.intel.com/content/www/us/en/products/docs/processors/xeon-accelerated/4th-gen-xeon-scalable-processors.html) (codenamed Sapphire Rapids). The toolkit provides the below key features:

* Seamless user experience of model compressions on Transformer-based models by extending [Hugging Face transformers](https://github.com/huggingface/transformers) APIs and leveraging [Intel® Neural Compressor](https://github.com/intel/neural-compressor)
* Advanced software optimizations and unique compression-aware runtime.
* Optimized Transformer-based model packages.
* [NeuralChat](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/neural_chat), a customizable chatbot framework to create your own chatbot within minutes by leveraging a rich set of plugins and SOTA optimizations.
* [Inference](https://github.com/intel/intel-extension-for-transformers/blob/main/intel_extension_for_transformers/llm/runtime/graph) of Large Language Model (LLM) in pure C/C++ with weight-only quantization kernels.
This PR is an integration of weight only quantization feature with intel-extension-for-transformers.

Unit test is in lib/langchain/tests/integration_tests/llm/test_weight_only_quantization.py
The notebook is in docs/docs/integrations/llms/weight_only_quantization.ipynb.
The document is in docs/docs/integrations/providers/weight_only_quantization.mdx.

### CODE REVIEW

**Review by hwchase17 — COMMENTED** [review]
we are moving to put integrations in langchain_community... should be a pretty straightfoward move, but lmk if any questions!

**Comment by PenghuiCheng** [comment]
> we are moving to put integrations in langchain_community... should be a pretty straightfoward move, but lmk if any questions!

It‘s OK for me.

**Comment by PenghuiCheng** [comment]
> we are moving to put integrations in langchain_community... should be a pretty straightfoward move, but lmk if any questions!

Hi, @hwchase17 , Who can move the code to langchain_community?

**Comment by PenghuiCheng** [comment]
Hi, @hwchase17 , I had moved code into langchain_community. Please review it, thanks very much!

File: docs/docs/integrations/providers/weight_only_quantization.mdx
Comment (baskaryan) [inline docs/docs/integrations/providers/weight_only_quantization.mdx:1]: should we start a more general intel.mdx page instead and move this there?

**Comment by PenghuiCheng** [comment]
@baskaryan , How can I veiw the error log for deploying on Vercel?

**Comment by PenghuiCheng** [comment]
@baskaryan， could you review the PR? thanks!

File: libs/langchain/langchain/llms/__init__.py
Comment (baskaryan) [inline libs/langchain/langchain/llms/__init__.py:1]: can we move any changes to libs/langchain, these imports are only here for backwards compatibility and that isn't needed for new integrations

Comment (PenghuiCheng) [reply libs/langchain/langchain/llms/__init__.py:1]: In libs/langchain, there is no __init_-.py file.

Comment (PenghuiCheng) [reply libs/langchain/langchain/llms/__init__.py:1]: @baskaryan , I removed changes in libs/langchain/langchain/llm/__init__.py, is it OK for you? thanks very much for your review.

**Review by baskaryan — APPROVED** [review]

**Comment by hshen14** [comment]
Hi @baskaryan, thanks for your review and approval. Please let us know whether we should invite some other reviewers to make the PR move forward.

**Comment by PenghuiCheng** [comment]
hi, @baskaryan , ITREX provides a package for Python 3.8, Please review the PR, thanks very much!

**Comment by hshen14** [comment]
Hi @baskaryan Thanks for your review. Please let us know whether you have additional comments.

File: docs/docs/integrations/llms/weight_only_quantization.ipynb
Comment (baskaryan) [inline docs/docs/integrations/llms/weight_only_quantization.ipynb:1]: notebook seems to have an error

Comment (PenghuiCheng) [reply docs/docs/integrations/llms/weight_only_quantization.ipynb:1]: @baskaryan , The issue of the notebook seems to not be related to this PR.

**Pull request merged**